src/lib/openapi-to-json-schema.ts

```diff
@@ -282,8 +323,9 @@ export default function toJSONSchema(
       if ('mapping' in schema.discriminator && typeof schema.discriminator.mapping === 'object') {
         // Discriminator mappings aren't written as traditional `$ref` pointers so in order to log them to the supplied
         // `refLogger`.
-        Object.keys(schema.discriminator.mapping).forEach(k => {
-          refLogger(schema.discriminator.mapping[k]);
+        const mapping = schema.discriminator.mapping;
+        Object.keys(mapping).forEach(k => {
+          refLogger(mapping[k]);
```

Comment:
Had to make this change because TS didn't like how we were accessing `schema.discriminator` here and didn't think that `discriminator` was present on the `schema` type (despite the same code working two lines above).